api/src/modules/import-data/sourcing-records/import.service.ts

```diff
@@ -85,11 +85,13 @@ export class SourcingRecordsImportService {
   ): Promise<void | Array<ErrorConstructor>> {
     const validationErrorArray: Array<typeof Error> = [];
     for (const parsedSheet in dtoLists) {
-      for (const dto of dtoLists[parsedSheet as keyof SourcingRecordsDtos]) {
-        try {
-          await validateOrReject(dto);
-        } catch (err) {
-          validationErrorArray.push(err);
+      if (dtoLists.hasOwnProperty(parsedSheet)) {
```

Comment:
@tiagojsag  the purpose of this clause is to avoid the `for of` loop to look down through all the prototype chain, which we totally sure we don't want / need